### PR TITLE
Servo fixes

### PR DESCRIPTION
**Pull request by supertick**

### CODE REVIEW

File: src/test/resources/ProgramAB/bots/pikachu/aiml/pikachu.aiml [hunk 1]
```diff
@@ -1,10 +1,8 @@
...
-		<!--  あなたを*に会うのはいいことだ   ?? -->
-		<!--  あなたを*に会えてうれしい -->
-		<!-- あなたに会えてよかったXXXXX -->
-		<template>あなたに会えてよかった<star /></template>
-	</category>
-</aiml>
\ No newline at end of file
+<?xml version="1.0" encoding="UTF-8"?>
+<aiml>
+<!--  -->
```
Comment (kwatters) [inline src/test/resources/ProgramAB/bots/pikachu/aiml/pikachu.aiml]: please revert this change.

Comment (kwatters) [reply src/test/resources/ProgramAB/bots/pikachu/aiml/pikachu.aiml]: programab should no longer overwrite this file when the test runs..

File: .gitignore [hunk 2]
```diff
@@ -98,3 +98,4 @@ Thumbs.db
 /models/
 /Common/
 /bulletjme.dll
+testbots/*
```
Comment (kwatters) [inline .gitignore]: this entry shouldn't be necessary

File: src/test/resources/ProgramAB/bots/pikachu/aiml/pikachu.aiml (hunk 1 above)
Comment (supertick) [reply src/test/resources/ProgramAB/bots/pikachu/aiml/pikachu.aiml]: reverted to develop version

File: .gitignore (hunk 2 above)
Comment (supertick) [reply .gitignore]: removed

File: src/main/java/org/myrobotlab/service/Emoji.java [hunk 3]
```diff
@@ -61,111 +61,11 @@
...
-    log.info(emotionalState.getFsmMap());
-
-    // loopback on state publishing...
-    // subscribe(getName(),"publishState");
-    try {
-      emotionalState.attach(this);
-    } catch (Exception e) {
-      error(e);
-    }
+    startPeer("fsm");
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/Emoji.java]: we really need to call this 3 times?

File: src/test/java/org/myrobotlab/document/transformer/AbstractStageTest.java [hunk 4]
```diff
@@ -13,17 +13,18 @@
...
-    stage.processDocument(doc);
-    validate(doc);
+  @Before
+  public void init() {
+    // LoggingFactory.init("WARN");
+  }
+
+  @Test
+  public void test() {
+    if (printMethods)System.out.println(String.format("Running %s.%s", getSimpleName(), getName()));
```
Comment (kwatters) [inline src/test/java/org/myrobotlab/document/transformer/AbstractStageTest.java]: Is this necessary now?

File: src/test/java/org/myrobotlab/integration/InMoovScriptTest.java [hunk 5]
```diff
@@ -115,6 +119,7 @@ public void testInMoovMinimalHead() throws IOException {
   // Test the inmoov minimal arm script.
   @Test
   public void testInMoovMinimalTorso() throws IOException {
+    if (printMethods)System.out.println(String.format("Running %s.%s", getSimpleName(), getName()));
```
Comment (kwatters) [inline src/test/java/org/myrobotlab/integration/InMoovScriptTest.java]: this sort of debug message should be refactored into the base class... or removed altogether.

File: src/main/java/org/myrobotlab/service/interfaces/PinDefinition.java [hunk 6]
```diff
@@ -5,9 +5,11 @@
 import com.pi4j.io.gpio.GpioPinDigitalMultipurpose;
 
 public class PinDefinition extends SensorDefinition implements Serializable {
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/interfaces/PinDefinition.java]: i think we should rename this class to be "Pin"

File: src/main/java/org/myrobotlab/service/interfaces/EncoderControl.java [hunk 7]
```diff
@@ -11,8 +11,21 @@
 
   // TODO: publish encoder data that includes the name/id of the device.
   public void onEncoderData(EncoderData data);
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/interfaces/EncoderControl.java]: shouldn't an encoder control publishEncoderData , not subscribe to it?

File: src/main/java/org/myrobotlab/service/interfaces/Microcontroller.java [hunk 8]
```diff
@@ -25,4 +25,8 @@
    * @return list of board types
    */
   public List<BoardType> getBoardTypes();
+  
+  public void reset();
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/interfaces/Microcontroller.java]: awesome.... yes, we should be able to reset microcontrollers...
